src/test/java/api/loans/InHouseUseCheckInTest.java

```diff
@@ -0,0 +1,169 @@
+    final UUID checkInServicePoint = servicePointsFixture.cd4().getId();
+
+    final IndividualResource itemLocation = locationsFixture.basedUponExampleLocation(
+      builder -> builder.withPrimaryServicePoint(itemServicePoint)
+        .servedBy(checkInServicePoint));
+
+    final IndividualResource nod = itemsFixture.basedUponNod(
+      item -> item.withTemporaryLocation(itemLocation.getId()));
+
+    final CheckInByBarcodeResponse checkInResponse = loansFixture
+      .checkInByBarcode(nod, checkInServicePoint);
+
+    assertThat(checkInResponse.getJson().containsKey("loan"), is(false));
+    assertThat(checkInResponse.getJson().containsKey("item"), is(true));
+    assertThat(checkInResponse.getInHouseUse(), is(true));
+  }
+
+  @Test
+  public void setInHouseUseFalseIfItemNotAvailable() {
+    final UUID checkInServicePointId = servicePointsFixture.cd1().getId();
+
+    final IndividualResource homeLocation = locationsFixture.basedUponExampleLocation(
+      builder -> builder.withPrimaryServicePoint(checkInServicePointId));
+
+    final IndividualResource nod = itemsFixture.basedUponNod(
+      item -> item
+        .withTemporaryLocation(homeLocation.getId()));
+
+    final IndividualResource checkOutResponse = loansFixture
+      .checkOutByBarcode(nod, usersFixture.james());
+
+    final String itemStatus = checkOutResponse.getJson().getJsonObject("item")
+      .getJsonObject("status").getString("name");
+    assertThat(itemStatus, is("Checked out"));
+
+    final CheckInByBarcodeResponse checkInResponse = loansFixture
+      .checkInByBarcode(nod, checkInServicePointId);
+
+    assertThat(checkInResponse.getJson().containsKey("loan"), is(true));
+    assertThat(checkInResponse.getJson().containsKey("item"), is(true));
+    assertThat(checkInResponse.getInHouseUse(), is(false));
+  }
+
+  @Test
+  public void setInHouseUseFalseIfItemHasRequest() {
+    final UUID checkInServicePointId = servicePointsFixture.cd1().getId();
+
+    final IndividualResource homeLocation = locationsFixture.basedUponExampleLocation(
+      builder -> builder.withPrimaryServicePoint(checkInServicePointId));
+
+    final IndividualResource nod = itemsFixture.basedUponNod(
+      item -> item
+        .withTemporaryLocation(homeLocation.getId()));
+
+    requestsFixture.place(new RequestBuilder()
+      .page()
+      .forItem(nod)
+      .withPickupServicePointId(checkInServicePointId)
+      .by(usersFixture.james()));
+    assertThat(requestsFixture.getQueueFor(nod).getTotalRecords(), is(1));
+
+    final CheckInByBarcodeResponse checkInResponse = loansFixture
+      .checkInByBarcode(nod, checkInServicePointId);
+
+    assertThat(checkInResponse.getJson().containsKey("loan"), is(false));
+    assertThat(checkInResponse.getJson().containsKey("item"), is(true));
+    assertThat(checkInResponse.getInHouseUse(), is(false));
+  }
+
+  @Test
+  public void setInHouseUseFalseIfCheckInServicePointDoNotMatch() {
```

Comment:
What should the check in service point not match?

Reply:
Check-in service point ID should be either item.location.primaryServicePointId or in item.location.servicePointIds array.